src/components/RiskComponents.vue

```diff
@@ -1,84 +1,162 @@
+        </RiskComponent>
       </v-col>
     </v-row>
-    <v-row no-gutters v-else>
-      <v-col cols="12" sm-and-down>
+    <!-- displays on small and down breakpoints -->
+    <v-row no-gutters class="hidden-md-and-up">
+      <v-col cols="12">
         <v-expansion-panels flat>
           <RiskComponentDropdown
-            :activity="activity"
-            type="crowding"
-          ></RiskComponentDropdown>
-          <RiskComponentDropdown
-            :activity="activity"
-            type="droplets"
-          ></RiskComponentDropdown>
-          <RiskComponentDropdown
-            :activity="activity"
-            type="exposureTime"
-          ></RiskComponentDropdown>
-          <RiskComponentDropdown
-            :activity="activity"
-            type="ventilation"
-          ></RiskComponentDropdown>
+            v-for="(risk, index) in riskData"
+            :key="risk.type"
+            :class="index && 'top-border'"
+            v-bind="risk"
+          >
+            <template #icon>
+              <component
+                :is="icons[risk.type]"
+                class="component-icon dropdown-icon"
+                :class="`${risk.riskClass}`"
+              />
+            </template>
+            <template #notes>
+              <Markdown :source="risk.notes" class="notes dropdown-notes" />
+            </template>
+          </RiskComponentDropdown>
         </v-expansion-panels>
       </v-col>
     </v-row>
   </v-container>
 </template>
 
 <script>
+// getters
+import { mapGetters } from "vuex";
+// components
 import RiskComponent from "@/components/RiskComponent.vue";
 import RiskComponentDropdown from "@/components/RiskComponentDropdown.vue";
+// icons
+import CrowdingIcon from "@/assets/Risk_Crowding-filled.svg";
+import DropletsIcon from "@/assets/Risk_Droplets-filled.svg";
+import TimeIcon from "@/assets/Risk_Time-filled.svg";
+import VentIcon from "@/assets/Risk_Ventilation-filled.svg";
+// markdown
+import Markdown from "vue-markdown";
 
 export default {
   components: {
     RiskComponent,
-    RiskComponentDropdown
+    RiskComponentDropdown,
+    Markdown
   },
   props: {
     activity: Object
   },
+  data: () => ({
+    riskTypes: ["crowding", "droplets", "exposureTime", "ventilation"],
+    riskLabels: {
+      "1": "Low",
+      "2": "Medium",
+      "3": "High"
+    },
+    icons: {
+      crowding: CrowdingIcon,
+      droplets: DropletsIcon,
+      exposureTime: TimeIcon,
+      ventilation: VentIcon
+    }
+  }),
   computed: {
+    ...mapGetters(["components"]),
     hasRiskData() {
-      return (
-        this.activity.dropletsNotes &&
-        this.activity.exposureTimeNotes &&
-        this.activity.crowdingNotes &&
-        this.activity.ventilationNotes
-      );
-    },
-    maybeSeparator() {
-      return this.$vuetify.breakpoint.mdAndUp ? "left-border" : "top-border";
+      for (const risk of this.riskTypes) {
+        if (!this.activity?.[`${risk}Notes`]) return false;
```

Comment:
For a while I had a lot of these guard `.?` things all over, but now I'm taking a different approach -- this component should never be displayed if it isn't provided a valid activity. Previously I ran into this a lot because of using `v-show` when I should have been using `v-if` in the parent component.

Reply:
@awwaiid also the `?.` is an [optional chaining operator](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Optional_chaining). 
basically it keeps prevents errors from being thrown when trying to access a property on something that's nullish (in this case, it would be `activity`). if `activity` was nullish, the evaluation would return `undefined` which is falsy value. so this is basically equivalent to the old version of `hasRiskData`, but with a little bit of an extra safety net